reporting-app/app/controllers/certifications_controller.rb

```diff
@@ -78,10 +78,17 @@ def certification_params
         begin
           # handle HTML form input of the JSON blob as a string
           if cert_params[:certification_requirements].present? && cert_params[:certification_requirements].is_a?(String)
+            parsed_requirements = JSON.parse(cert_params[:certification_requirements])
+
+            # Permit all Requirements and RequirementParams attributes, plus nested params hash
+            permitted_keys = (Certifications::Requirements.attribute_names | Certifications::RequirementParams.attribute_names).map(&:to_sym)
+
             cert_params[:certification_requirements] =
-              ActionController::Parameters.new(
-                JSON.parse(cert_params[:certification_requirements])
-              ).permit((Certifications::Requirements.attribute_names | Certifications::RequirementParams.attribute_names).map(&:to_sym))
+              ActionController::Parameters.new(parsed_requirements).permit(
+                *permitted_keys,
+                months_that_can_be_certified: [],
+                params: permitted_keys
```

Comment:
After looking at the params more closely I think we could do something like:

```suggestion
                *Certifications::Requirements.attribute_names.map(&:to_sym).excluding(:months_that_can_be_certified, :params),
                months_that_can_be_certified: [],
                params: Certifications::RequirementParams.attribute_names.map(&:to_sym)
```

Then we can remove setting `permitted_keys` above